packages/govuk-frontend/src/govuk/components/skip-link/skip-link.mjs

```diff
@@ -31,12 +31,40 @@ export class SkipLink extends GOVUKFrontendComponent {
 
     this.$module = $module
 
-    const linkedElementId = getFragmentFromUrl(this.$module.hash)
+    const hash = this.$module.hash
+    const href = this.$module.getAttribute('href') ?? ''
+
+    /** @type {URL | undefined} */
+    let url
+
+    /**
+     * Check for valid link URL
+     *
+     * {@link https://caniuse.com/url}
+     * {@link https://url.spec.whatwg.org}
+     *
+     */
+    try {
+      url = new window.URL(this.$module.href)
+    } catch (error) {
+      throw new ElementError(
+        `Skip link: Target link (\`href="${href}"\`) is invalid`
+      )
+    }
+
+    // Check for external link URL and return early
+    if (url.origin !== window.location.origin) {
+      return
+    }
+
+    const linkedElementId = getFragmentFromUrl(hash)
 
     // Check for link hash fragment
-    if (!linkedElementId) {
+    if (!linkedElementId || url.pathname !== window.location.pathname) {
```

Comment:
Given we switch which error we throw based on `linkedElementId` anyway, I think it might make be more readable to split this into two separate throws?

```mjs
if (url.pathname !== window.location.pathname) {
  throw new ElementError(
    `Skip link: Target link (\`href="${href}"\`) must stay on page (\`${window.location.pathname}\`)`
  )
}

const linkedElementId = getFragmentFromUrl(hash)

if (!linkedElementId) {
  throw new ElementError(
    `Skip link: Target link (\`href="${href}"\`) has no hash fragment`
  )
}
```

Reply:
Actually, I think we might need to take another look at this because of the reasons you mentioned on Slack earlier – from [our guidance](https://design-system.service.gov.uk/components/exit-this-page/#consider-what-to-do-with-user-session-data):

> You’ll need to decide if your service will store the user’s session data after they’ve activated the Exit this page component and left your service.
> 
> For example, you could do this in 2 steps:
>
> 1. replace the default website link to a URL in your service, that when visited, will clear the user’s session data
> 2. redirect the user to another website, such as BBC Weather

In the case of option 1, the URL used could be a different pathname on the same origin – e.g. `/clear-session-data`. In that instance I think we want this to return silently without throwing, the same way we do if the origin doesn't match.

Reply:
What if we check for the presence of a hash fragment and keep the error?

But for paths like `/clear-session-data` without a hash fragment we return early

Just tried it with https://github.com/alphagov/govuk-frontend/pull/4536/commits/f93a017aba26ed1ed5823399e7c74989e77df7f1

Reply:
Just caught up with @romaricpascal

>…we’re thinking it’d be less risky to go with only throwing if it’s a same page link and there’s no hash, otherwise early returning and not worrying, rather than trying to be too tight around expecting a link to the same page specifically.

So I've pushed 5cf5c3289a4ab85b2360bcff763fa9404cc58062 with a single early return

```mjs
// Return early for external URLs or links to other pages
if (
  url.origin !== window.location.origin ||
  url.pathname !== window.location.pathname
) {
  return
}
```

Which means this error message is no longer included too:

```console
Target link (`href="/somewhere-else#main-content"`) must stay on page (`/components/skip-link/preview`)
```

Reply:
Sounds good to me 👍🏻